jvm/workbookapp/src/main/kotlin/org/wycliffeassociates/otter/jvm/workbookapp/oqua/ChapterViewModel.kt

```diff
@@ -109,23 +148,43 @@ class ChapterViewModel : ViewModel() {
         }
     }
 
-    private fun closeChapterAudio() {
+    private fun closeAudio() {
         audioPlayerProperty.value.close()
     }
 
     private fun saveDraftReview() {
         draftReviewRepo.writeDraftReviewFile(workbook, chapterNumber, questions)
     }
 
-    fun jumpToVerse(verse: Int) {
-        if (verse > 0 && verse <= verseMarkerModel.markers.size) {
-            val frame = getVerseFrame(verse)
-            audioPlayerProperty.value.seek(frame)
-            audioPlayerProperty.value.play()
+    fun playVerseRange(start: Int, end: Int) {
+        if (hasAllMarkers.value) {
+            if ((start in 1..numberOfVerses) &&
+                (end in start..numberOfVerses)
+            ) {
+
+                val startFrame = getVerseFrame(start)
+                val endFrame = getVerseEndFrame(end)
+
+                audioPlayerProperty.value.loadSection(take.file, startFrame, endFrame)
+                audioPlayerProperty.value.seek(0)
+                audioPlayerProperty.value.play()
+            } else {
+                throw IndexOutOfBoundsException(
+                    "ChapterViewModel: Verse range [$start - $end] does not exist in $numberOfVerses verses"
+                )
+            }
+        }
+    }
+
+    private fun getVerseEndFrame(verse: Int): Int {
+        return if (verse == numberOfVerses) {
+            numberOfFrames
```

Comment:
can we just do `audioPlayerProperty.value.getDurationInFrames()` here instead of having a dedicated property for this only place?

Reply:
Unfortunately no because getDurationInFrames returns the length of the loaded section. I am loading different sections at different times. So when I want the whole audio's length, I need to save it initially because it won't be accessible later.

Reply:
oh okay, then you should name it something like `totalFrames` to avoid confusion with the current duration